src/vmemcache_repl.c

```diff
@@ -183,6 +196,18 @@ repl_p_lru_new(struct repl_p_head **head)
 static void
 repl_p_lru_delete(struct repl_p_head *head)
 {
+	util_mutex_lock(&head->lock);
+
+	while (!STAILQ_EMPTY(&head->first)) {
+		struct repl_p_entry *entry = STAILQ_FIRST(&head->first);
+		STAILQ_REMOVE_HEAD(&head->first, node);
+		Free(entry);
+	}
+
+	util_mutex_unlock(&head->lock);
+	util_mutex_destroy(&head->lock);
```

Comment:
If you need to take a lock before destroy, something is seriously wrong.

Reply:
.